sktime/transformations/series/compose.py

```diff
@@ -430,3 +433,145 @@ def _check_is_pdseries(z):
+    >>> y_train, y_test, X_train, X_test = temporal_train_test_split(y, X)
+    >>> fh = ForecastingHorizon(y_test.index, is_relative=False)
+
+    >>> # we want to fit the Imputer only on the predict (=transform) data.
+    >>> # note that NaiveForecaster cant use X data, this is just a show case.
+    >>> pipe = ForecastingPipeline(
+    ...     steps=[
+    ...         ("imputer", FitInTransform(Imputer(method="mean"))),
+    ...         ("forecaster", NaiveForecaster()),
+    ...     ]
+    ... )
+    >>> pipe.fit(y_train, X_train)
+    ForecastingPipeline(...)
+    >>> y_pred = pipe.predict(fh=fh, X=X_test)
+    """
+
+    def __init__(self, transformer):
+        self.transformer = transformer
+        super(FitInTransform, self).__init__()
+        self.clone_tags(transformer, None)
+        self.set_tags(**{"fit_is_empty": True})
+        # self.set_tags(handles-missing-data=True)
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        private _transform containing core logic, called from transform
+
+        Parameters
+        ----------
+        X : Series or Panel of mtype X_inner_mtype
+            if X_inner_mtype is list, _transform must support all types in it
+            Data to be transformed
+        y : Series or Panel of mtype y_inner_mtype, default=None
+            Additional data, e.g., labels for transformation
+
+        Returns
+        -------
+        transformed version of X
+        """
+        self.transformer_ = clone(self.transformer)
+        self.transformer_.fit(X=X, y=y)
+        return self.transformer_.transform(X=X, y=y)
+
+    def _inverse_transform(self, X, y=None):
+        """Inverse transform, inverse operation to transform.
+
+        private _inverse_transform containing core logic, called from inverse_transform
+
+        Parameters
+        ----------
+        X : Series or Panel of mtype X_inner_mtype
+            if X_inner_mtype is list, _inverse_transform must support all types in it
+            Data to be inverse transformed
+        y : Series or Panel of mtype y_inner_mtype, optional (default=None)
+            Additional data, e.g., labels for transformation
+
+        Returns
+        -------
+        inverse transformed version of X
+        """
+        return self.transformer_.inverse_transform(X=X, y=y)
+
+    def _update(self, X, y=None):
+        """Update transformer with X and y.
+
+        private _update containing the core logic, called from update
+
+        Parameters
+        ----------
+        X : Series or Panel of mtype X_inner_mtype
+            if X_inner_mtype is list, _update must support all types in it
+            Data to update transformer with
+        y : Series or Panel of mtype y_inner_mtype, default=None
+            Additional data, e.g., labels for tarnsformation
+
+        Returns
+        -------
+        self: reference to self
+        """
+        self.transformer_.update(X=X, y=y)
```

Comment:
I think `update` should be empty. The idea is for this to be a "stateless" version that is always applied to the batch.